dali/pipeline/operators/decoder/nvjpeg_decoder_decoupled_api.h

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_DECODER_DECOUPLED_API_H_
+#define DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_DECODER_DECOUPLED_API_H_
+
+#include <nvjpeg.h>
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/operators/decoder/nvjpeg_helper.h"
+#include "dali/util/image.h"
+#include "dali/util/ocv.h"
+#include "dali/image/image_factory.h"
+#include "dali/pipeline/util/thread_pool.h"
+
+namespace dali {
+
+class nvJPEGDecoder : public Operator<MixedBackend> {
+ public:
+  explicit nvJPEGDecoder(const OpSpec& spec) :
+    Operator<MixedBackend>(spec),
+    output_image_type_(spec.GetArgument<DALIImageType>("output_type")),
+    hybrid_huffman_threshold_(spec.GetArgument<unsigned int>("hybrid_huffman_threshold")),
+    output_info_(batch_size_),
```

Comment:
Please keep the order of initialization list and order of member declaration consistent

Reply:
It is consistent, gcc is throwing an error when it isn't